src/framework/audio/internal/worker/mixer.h

```diff
@@ -24,60 +24,62 @@
 
 #include <memory>
 #include <map>
+
+#include "modularity/ioc.h"
+#include "async/asyncable.h"
+
 #include "imixer.h"
+#include "iaudioconfiguration.h"
 #include "abstractaudiosource.h"
 #include "mixerchannel.h"
 #include "clock.h"
 
 namespace mu::audio {
-class Mixer : public IMixer, public AbstractAudioSource, public std::enable_shared_from_this<Mixer>
+class Mixer : public IMixer, public AbstractAudioSource, public std::enable_shared_from_this<Mixer>, public async::Asyncable
 {
+    INJECT(audio, IAudioConfiguration, config)
```

Comment:
for a mixer and other similar classes that should be in an audio worker, I would recommend setting options explicitly, rather than using dependencies.
Otherwise, we need to make sure that AudioConfiguration (and other dependencies) are thread-safe.